pkg/plan/resource/kubeadm_init.go

```diff
@@ -251,13 +251,29 @@ func buildKubeadmInitPlan(path string, ignorePreflightErrors string, useIPTables
 		uploadCertsFlag = "--experimental-upload-certs"
 	}
 
+	// If we're at 1.17.0 or greater, we need to upgrade the kubeadm config before running "kubeadm init"
+	upgradeKubeadmConfig := false
+	if lt, err := version.LessThan(k8sVersion, "1.17.0"); err == nil && lt == false {
+		upgradeKubeadmConfig = true
+	}
+
 	b := plan.NewBuilder()
 	if useIPTables {
 		b.AddResource(
 			"configure:iptables",
 			&Run{Script: object.String("sysctl net.bridge.bridge-nf-call-iptables=1")}) // TODO: undo?
 	}
 
+	if upgradeKubeadmConfig {
+		b.AddResource(
+			"kubeadm:config:upgrade",
+			&Run{Script: plan.ParamString(
```

Comment:
This is a really minor style point. We've mostly tried in the past to use `ParamString` only for substituting values that aren't available until runtime. This lets readers know that there is communication going on at runtime between different resources. I notice there's another instance of this kind of use of `ParamString` below as well. The pattern is normally to use `object.String(fmt.Sprintf(...` for substitutions available at plan creation time like this example from `install_plans.go`: 

```object.String(fmt.Sprintf("echo %q | tee -a %q", repoLine, repoFile))```